tuna/miopen/scripts/report.py

```diff
@@ -0,0 +1,251 @@
+  """Check for configs that are in the session tuning but not in the golden_v tuning"""
+  #get number of configs that are in golden_v but not in session
+
+  sess_configs = set(session_data['config'].unique().flatten().tolist())
+  golden_configs = set(golden_data['config'].unique().flatten().tolist())
+
+  sess_miss = sess_configs.difference(golden_configs)
+  gv_miss = golden_configs.difference(sess_configs)
+
+  print_driver_cmds(args, dbt, list(sess_miss), "Missing commands", 'session')
+  print_driver_cmds(args, dbt, list(gv_miss), "Missing commands", 'conv_golden')
+
+
+def print_driver_cmds(args, dbt, ids_list, text, table1):
+  """Print configs present in @table1 but missing from @table1"""
+  with DbSession() as session:
+    query = session.query(dbt.config_table.id, dbt.config_table.driver)\
+                   .filter(dbt.config_table.id.in_(ids_list))
+    drivers = query.all()
+    LOGGER.info("%s from %s:  %s", text, table1, len(drivers))
+    LOGGER.info("Driver cmds written to missing_%s_sess%s_gv%s.txt", table1,
+                args.session_id, args.golden_v)
+    missing_drivers = []
+    with open(f"missing_{table1}_sess{args.session_id}_gv{args.golden_v}.txt",
+              'w',
+              encoding='utf-8') as fout:
+      for entry in drivers:
+        if entry[1] is not None:
+          fout.write(f"{entry[1]}\n")
+        else:
+          missing_drivers.append(entry[0])
+    if missing_drivers:
+      LOGGER.warning(
+          'Configs with missing drivers in db: %s that could not be written to file',
+          len(missing_drivers))
+
+
+def configs_report(args, dfr):
+  """Print tuning summary"""
+
+  #remove entries where sess or gv dont have an entry for the same config
+  dfr = dfr.loc[~dfr[['solver_x', 'solver_y']].isna().any(axis=1)].copy()
+  #compute ktime diff for fastest solver/config
+  dfr['diff'] = dfr['ktime_x'] - dfr['ktime_y']
+
+  #prct configs faster or slower
+  prct_positive = (dfr['diff'] > 0).sum() / dfr.shape[0] * 100
+  prct_equal = (dfr['diff'] == 0).sum() / dfr.shape[0] * 100
+  prct_negative = (dfr['diff'] < 0).sum() / dfr.shape[0] * 100
+  #pylint: disable=logging-format-truncated
+  LOGGER.info("configs with faster kernel_time: %f %%", round(prct_negative, 4))
+  LOGGER.info("configs with equal kernel_time: %f %%", round(prct_equal, 4))
+  LOGGER.info("configs with slower kernel_time: %f %% \n",
+              round(prct_positive, 4))
+
+  #averages
+  avg_positive = (dfr['diff'] > 0).mean()
+  avg_negative = (dfr['diff'] < 0).mean()
+
+  LOGGER.info("Mean for configs with faster kernel_time: %s %%", avg_negative)
+  LOGGER.info("Mean for configs with slower kernel_time: %s %%", avg_positive)
+  LOGGER.info("Mean for all configs: %s %%", dfr['diff'].mean())
+
+  dfr['%speedup'] = (dfr['ktime_x'] - dfr['ktime_y']) / (
+      (dfr['ktime_x'] + dfr['ktime_y']) / 2) * 100
+  LOGGER.info("Overall speed-up: %s %%", round(dfr['%speedup'].mean(), 4))
+
```

Comment:
You are comparing the difference of ktime_x and ktime_y to the avg of ktime_x and ktime_y -- which  doens't seem like the right thing to do. Instead, do
```
dfr['%speedup'] = (dfr['ktime_y'] - dfr['ktime_x']) / dfr['ktime_y'] * 100
```
Notice that, here, %speedup will be +ve if ktime_y is greater than ktime_x (i.e., if the solvers in golden_data are slower than the session_data)

Reply:
Thanks for that, I wasnt sure which formula to use

Reply:
@alexandraBara I see, maybe you want to flip it to y-x/y*100 everywhere? Because as is, -ve numbers in your report actually mean something positive (i.e., a speed-up). So
- `Overall speed-up: -5%`,  means speed-up
- `Overall speed-up: +5%`,  means slow-down

Reply:
@cderb @JehandadKhan  what makes more sense? -5% as speed-up or +5% as speedup?